cmd/cli/version_test.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/pkg/errors"
+	tassert "github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+type fakeRemoteVersion struct {
+	err         error
+	versionInfo *version.Info
+}
+
+func (f *fakeRemoteVersion) proxyGetMeshVersion(pod string, namespace string, clientset kubernetes.Interface) (*version.Info, error) {
+	if f.err != nil {
+		return nil, f.err
+	}
+	return f.versionInfo, nil
```

Comment:
I suppose this could be simplified a bit since the current test cases only define one of these anyway.
```suggestion
	return f.versionInfo, f.err
```